### PR TITLE
fix(android/engine): Check suggestion banner after setting keyboard

### LINKED ISSUE #6554: bug(android/engine): Blank suggestion banner after install keyboard package
Found while investigating #4936 

In the latest 15.0 beta, after installing a keyboard package (that has no associated lexical-model), the keyboard switch to the new keyboard is showing a blank suggestion banner.

Steps to repro:
1. Load the latest 15.0 beta
2. In the Keyman "Settings" menu, install a keyboard (e.g. khmer_angkor)
3. After khmer_angkor installs, observe current keyboard is khmer_angkor with a blank suggestion banner

### PR DESCRIPTION
**Pull request by darcywong00**
Fixes #6554

The keyboard picker contains logic in `prepareKeyboardSwitch` to determine if the suggestion banner needs to be enabled/disabled. We need to do the same after installing a keyboard / setting new keyboard to determine if the suggestion banner needs to be enabled/disabled.

## User Testing
Setup
1. Load the PR build of "Keyman for Android"

* **TEST_BANNER_TOGGLE** - Tests suggestion banner is enabled/disabled
1. Start Keyman and dismiss the "Get Started" menu
2. Verify the default sil_euro_latin keyboard has suggestions visible
3. From Keyman Settings, search for and install a keyboard that doesn't have a dictionary (e.g. khmer_angkor)
4. Finish the keyboard package installation and verify the keyboard changes to the installed keyboard
5. Verify the suggestion banner is no longer visible
6. Longpress on the globe button to bring up the keyboard picker menu
7. Select sil_euro_latin
8. Verify the suggestion banner is visible

### CODE REVIEW

**Comment by bharanidharanj** [comment]
* **TEST_BANNER_TOGGLE (PASSED):** Tested this with the attached PR build in Android (11 Version) API 30 emulator and the banner only visible for EuroLatin (SIL) Keyboard.  (and not for Khmer Keyboard). Working as expected.

**Comment by jahorton** [comment]
Which step(s) of the specified user test previously would generate errors?  Step 5?

**Comment by darcywong00** [comment]
> Which step(s) of the specified user test previously would generate errors? Step 5?

Right, after installing the "khmer_angkor" keyboard, the blank suggestion banner would show.

**Review by rc-swag — APPROVED** [review]
lgtm

File: android/KMEA/app/src/main/java/com/tavultesoft/kmea/KMManager.java [hunk 1]
```diff
@@ -2126,6 +2116,21 @@ public static void setGlobeKeyState(GlobeKeyState state) {
     globeKeyState = state;
   }
 
+  private static void toggleSuggestionBanner(String languageID, boolean result1, boolean result2) {
```
Comment (mcdurdin) [inline android/KMEA/app/src/main/java/com/tavultesoft/kmea/KMManager.java:2119]: I don't really like the `result1`, `result2` names -- are these related to in-app vs system keyboard?

Comment (darcywong00) [reply android/KMEA/app/src/main/java/com/tavultesoft/kmea/KMManager.java:2119]: Yes. they're the respective statuses of `prepareKeyboardSwitch()` or `setKeyboard()`

`result1` for in-app keyboard
`result2` for system keyboard

Comment (jahorton) [reply android/KMEA/app/src/main/java/com/tavultesoft/kmea/KMManager.java:2119]: Is there a reason they can't be combined into a single boolean?  How would `false, false` and `true, true` be handled if not?

Comment (darcywong00) [reply android/KMEA/app/src/main/java/com/tavultesoft/kmea/KMManager.java:2119]: Are you recommending something like this?
```
if (result) {
  InAppKeyboard.setLayoutParams(getKeyboardLayoutParams());
  SystemKeyboard.setLayoutParams(getKeyboardLayoutParams());
}
```

Comment (jahorton) [reply android/KMEA/app/src/main/java/com/tavultesoft/kmea/KMManager.java:2119]: More like

```
if (result) {
  InAppKeyboard.setLayoutParams(getKeyboardLayoutParams());
} else {
  SystemKeyboard.setLayoutParams(getKeyboardLayoutParams());
}
```

Comment (darcywong00) [reply android/KMEA/app/src/main/java/com/tavultesoft/kmea/KMManager.java:2119]: Hmm - that doesn't handle the (often) scenario where both InApp and SystemKeyboard need to update `setLayoutParams()`.

Comment (jahorton) [reply android/KMEA/app/src/main/java/com/tavultesoft/kmea/KMManager.java]: > Hmm - that doesn't handle the (often) scenario where both InApp and SystemKeyboard need to update `setLayoutParams()`.

That sounds quite different from how the system keyboard works on iOS - are in-app and system keyboards both part of the same process on Android?  I'm curious how that's a scenario at all, let alone a frequent one.  (On iOS, it's only ever one or the other.  They're not in the same process.)

Comment (darcywong00) [reply android/KMEA/app/src/main/java/com/tavultesoft/kmea/KMManager.java]: In Keyman Engine for Android, all keyboard switch calls in KMManager are applied to both in-app and system at the same time.

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 15.0.242-beta](https://keyman.com/downloads/releases/beta/15.0.242)

**Pull request merged**